bridge-token-factory/src/lib.rs

```diff
@@ -372,21 +414,61 @@ impl BridgeTokenFactory {
             target, message
         ));
 
+        let token = EthAddressHex(token);
+        let fee_amount = self.calculate_deposit_fee_amount(
+            &token,
+            amount,
+            message.as_ref().and_then(|_| Some(&target)),
+        );
+        let amount_to_transfer = amount.checked_sub(fee_amount).unwrap_or(0);
+
+        env::log_str(&format!(
+            "Finish deposit. Target:{} Message:{:?}",
+            target, message
+        ));
+
         match message {
-            Some(message) => ext_bridge_token::ext(self.get_bridge_token_account_id(token.clone()))
-                .with_static_gas(MINT_GAS)
-                .with_attached_deposit(env::attached_deposit() - required_deposit)
-                .mint(env::current_account_id(), amount.into())
-                .then(
-                    ext_bridge_token::ext(self.get_bridge_token_account_id(token))
-                        .with_static_gas(FT_TRANSFER_CALL_GAS)
-                        .with_attached_deposit(1)
-                        .ft_transfer_call(target, amount.into(), None, message),
-                ),
-            None => ext_bridge_token::ext(self.get_bridge_token_account_id(token))
-                .with_static_gas(MINT_GAS)
-                .with_attached_deposit(env::attached_deposit() - required_deposit)
-                .mint(target, amount.into()),
+            Some(message) => {
+                let mint_self_promise =
+                    ext_bridge_token::ext(self.get_bridge_token_account_id(&token))
+                        .with_static_gas(MINT_GAS)
+                        .with_attached_deposit(self.bridge_token_storage_deposit_required)
```

Comment:
I don't like the logic with `storage_deposit`. Why the user should pay for the storage deposit for bridge factory? The Bridge Token registers the factory when it is created. We don't need to attach a deposit here.

Reply:
In the current implementation, we already attach this amount of deposit, so we can keep it as it was for backward compatibility. Let's cover this improvement in a follow-up PR as described below.